commons/Common.mli

```diff
@@ -140,6 +140,44 @@ val write_file : file:filename -> string -> unit
 *)
 val read_file : ?max_len:int -> filename -> string
 
+val is_directory : filename -> bool
+
+
+type path_special = private Path of string [@@deriving show, eq]
+
+module Path :
+sig
+  type t = path_special [@@deriving show, eq]
+  val of_string : string -> t
+  val to_string : t -> string
+  val canonical : string -> string
+
+  val (/) : t -> string -> t
+  val concat : t -> string -> t
+  val apply : f:(string -> 'a) -> t -> 'a
+
+  val cat : t -> string list
+
+  val write_file : file:t -> string -> unit
+
+  val read_file : ?max_len:int -> t -> string
+  val file_exists : t -> bool
+
+  val is_directory : t -> bool
+  val basename : t -> t
```

Comment:
This does not make sense to me for basename to return a t. I don't even understand how it could work.

Reply:
basename should return a string.